microservice/src/test/java/de/gsi/microservice/concepts/majordomo/MajordomoClientV2.java

```diff
@@ -0,0 +1,125 @@
+        reconnectToBroker();
+    }
+
+    /**
+     * Connect or reconnect to broker
+     */
+    void reconnectToBroker() {
+        if (clientSocket != null) {
+            clientSocket.close();
+        }
+        clientSocket = ctx.createSocket(SocketType.DEALER);
+        clientSocket.setHWM(0);
+        clientSocket.setIdentity("clientV2".getBytes(StandardCharsets.UTF_8));
+        clientSocket.connect(broker);
+        if (poller != null) {
+            poller.unregister(clientSocket);
+            poller.close();
+        }
+        poller = ctx.createPoller(1);
+        poller.register(clientSocket, ZMQ.Poller.POLLIN);
+        if (verbose) {
+            log.format("I: connecting to broker at %s...\n", broker);
+        }
+    }
+
+    /**
+     * Returns the reply message or NULL if there was no reply. Does not attempt
+     * to recover from a broker failure, this is not possible without storing
+     * all unanswered requests and resending them all…
+     */
+    public ZMsg recv() {
+        ZMsg reply = null;
+
+        // Poll socket for a reply, with timeout
+        if (poller.poll(timeout * 1000) == -1) {
+            return null; // Interrupted
+        }
+
+        if (poller.pollin(0)) {
+            ZMsg msg = ZMsg.recvMsg(clientSocket);
+            if (verbose) {
+                log.format("I: received reply: \n");
+                msg.dump(log.out());
+            }
+            // Don't try to handle errors, just assert noisily
+            assert (msg.size() >= 4);
+
+            ZFrame empty = msg.pop();
+            assert (empty.getData().length == 0);
+            empty.destroy();
+
+            ZFrame header = msg.pop();
+            assert (MdpSubProtocol.C_CLIENT.isEquals(header.getData()));
+            header.destroy();
+
+            ZFrame replyService = msg.pop();
+            replyService.destroy();
+
+            reply = msg;
+        }
+        return reply;
+    }
+
+    /**
+     * Send request to broker and get reply by hook or crook Takes ownership of request message and destroys it when sent.
+     *
+     * @param service UTF-8 encoded service name byte array
+     * @param msgs message(s) to be sent to MajordomoProtocol broker (if more than one, than the last is assumed to be a RBAC-token
+     */
+    public boolean send(final String service, final byte[]... msgs) {
+        return send(service.getBytes(StandardCharsets.UTF_8), msgs);
+    }
+
+    /**
+     * Send request to broker and get reply by hook or crook Takes ownership of request message and destroys it when sent.
+     *
+     * @param service UTF-8 encoded service name byte array
+     * @param msgs message(s) to be sent to MajordomoProtocol broker (if more than one, than the last is assumed to be a RBAC-token
+     */
+    public boolean send(final byte[] service, final byte[]... msgs) {
+        return MajordomoProtocol.sendClientMessage(clientSocket, MdpClientCommand.C_UNKNOWN, null, service, msgs);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'MajordomoProtocol.sendClientMessage' due to existing static import 'de.gsi.microservice.concepts.majordomo.MajordomoProtocol.*'](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6307742)